circle[] circles
int32 sizeX
int32 sizeY
